app/controllers/application_controller.rb

```diff
@@ -680,6 +681,19 @@ def check_xhr
     raise ApplicationController::RenderEmpty.new unless ((request.format && request.format.json?) || request.xhr?)
   end
 
+  def block_cdn_requests
+    raise Discourse::NotFound if Discourse.is_cdn_request?(request.env, request.method)
```

Comment:
I am not sure if I want this blocking, there really is no damage if some random request is made and there are just so many ways to configure a CDN I worry we will break something.

Can we simply keep the new headers without locking down the routes?

Reply:
Done. Now we only adding the new headers.